src/redis-cli.c

```diff
@@ -1142,6 +1143,69 @@ static sds cliFormatReplyCSV(redisReply *r) {
     return out;
 }
 
+static sds cliFormatReplyJson(redisReply *r) {
+    unsigned int i;
+
+    sds out = sdsempty();
+    switch (r->type) {
+    case REDIS_REPLY_ERROR:
+        out = sdscat(out,"error:");
+        out = sdscatrepr(out,r->str,strlen(r->str));
+        break;
+    case REDIS_REPLY_STATUS:
+        out = sdscatrepr(out,r->str,r->len);
+        break;
+    case REDIS_REPLY_INTEGER:
+        out = sdscatprintf(out,"%lld",r->integer);
+        break;
+    case REDIS_REPLY_DOUBLE:
+        out = sdscatprintf(out,"%s",r->str);
+        break;
+    case REDIS_REPLY_STRING:
+    case REDIS_REPLY_VERB:
+        out = sdscatrepr(out,r->str,r->len);
+        break;
+    case REDIS_REPLY_NIL:
+        out = sdscat(out,"null");
+        break;
+    case REDIS_REPLY_BOOL:
+        out = sdscat(out,r->integer ? "true" : "false");
+        break;
+    case REDIS_REPLY_ARRAY:
+    case REDIS_REPLY_SET:
+    case REDIS_REPLY_PUSH:
+        out = sdscat(out,"[");
+        for (i = 0; i < r->elements; i++ ) {
+            sds tmp = cliFormatReplyJson(r->element[i]);
```

Comment:
Maybe we could pass the output string as an argument and let the recursion keep catting to it, rather than copy it? 
I imagine some responses can be huge... 
In that sense, maybe buffering the entire thing is not wise and we could spit it out to stdout in bulks, but that's morr complicated...

Reply:
On the other hand, if the reply is huge we're already holding a huge `redisReply`.

Reply:
you mean that it'll already be inefficient?
still why do excessive copying?

Reply:
@yu74n do you wanna try the approach i suggested?
i.e. the recursion takes an sds as input (passing `sdsempty()` to the root), keeping `cat`ting into it, and returning the new pointer.

Reply:
Sure, let me try.